ui-admin/src/portal/PortalEnvConfigView.tsx

```diff
@@ -1,41 +1,92 @@
-import React from 'react'
-import { PortalEnvironment } from 'api/api'
+import React, { useState } from 'react'
+import Api, { PortalEnvironment } from 'api/api'
+import { useUser } from '../user/UserProvider'
+import _cloneDeep from 'lodash/cloneDeep'
+import { failureNotification, successNotification } from '../util/notifications'
+import { Store } from 'react-notifications-component'
+import { Portal } from '@juniper/ui-core/build/types/portal'
 
-// TODO: Add JSDoc
-// eslint-disable-next-line jsdoc/require-jsdoc
-const PortalEnvConfigView = ({ portalEnv }: {portalEnv: PortalEnvironment}) => {
-  const config = portalEnv.portalEnvironmentConfig
+type PortalEnvConfigViewProps = {
+  portal: Portal,
+  portalEnv: PortalEnvironment,
+  updatePortal: (portal: Portal) => void
+}
+
+/**
+ * allows viewing/editing of portal environment configs, e.g. password protection
+ */
+const PortalEnvConfigView = ({ portal, portalEnv, updatePortal }: PortalEnvConfigViewProps) => {
+  const [config, setConfig] = useState(portalEnv.portalEnvironmentConfig)
+  const { user } = useUser()
 
+  /** update a given field in the config */
+  const updateConfig = (propName: string, value: string | boolean) => {
+    const newConfig = _cloneDeep(config)
+    // eslint-disable-next-line
+    // @ts-ignore
+    newConfig[propName] = value
+    setConfig(newConfig)
```

Comment:
I've been liking Lodash FP for data munging like this since it treats values as immutable and returns copies.

```ts
import { set } from 'lodash/fp'

...

const updateConfig = (propName: string, value: string | boolean) => {
    setConfig(set(propName, value))
}
```

Reply:
oooh, that is a big improvement